tools/build.py

```diff
@@ -96,6 +96,8 @@ def devhelp(helpstring):
                          help=devhelp('build test version of the jerry command line tool (%(choices)s)'))
     compgrp.add_argument('--libfuzzer', metavar='X', choices=['ON', 'OFF'], type=str.upper,
                          help=devhelp('build jerry with libfuzzer support (%(choices)s)'))
+    compgrp.add_argument('--oss-fuzz', metavar='X', choices=['ON', 'OFF'], type=str.upper,
+                         help=devhelp('build jerry to run on OSS-fuzz (%(choices)s)'))
```

Comment:
I'm a bit unsure whether this is the right approach to configure the libfuzzer command line tool for oss-fuzz. E.g., `--oss-fuzz=on` makes no sense without `--libfuzzer=on`. I wonder whether the choices of `--libfuzzer` should be extended instead, e.g., to allow ON, OFF, and OSS. OFF and ON would keep their original meaning and OSS would be a "special ON" -- this could cover the three exclusive approaches with the existing cli option. (Plus, it removes the need for a new cmake option as well.)

Reply:
You are correct that fuzzing makes little sense without the sanitizers. In the case of OSS-fuzz, however, the flags will be set by OSS-fuzz itself to fuzz with other sanitizers, and forcing a specific sanitizer on the compilation will be incorrect.

Reply:
Nah, that's not what I was trying to point out. In the currently suggested form, the cmake has:
```cmake
if(JERRY_LIBFUZZER)
  jerry_create_executable("jerry-libfuzzer" "libfuzzer.c")
  if(JERRY_OSS_FUZZ)
    target_link_libraries("jerry-libfuzzer" jerry-port-default)
  else()
    target_link_libraries("jerry-libfuzzer" jerry-port-default -fsanitize=fuzzer)
  endif()
endif()
```

That is, if `JERRY_LIBFUZZER` is falsy, then the value of `JERRY_OSS_FUZZ` is indifferent. So, `JERRY_OSS_FUZZ` is not an independent configuration variable. The valid combination are `JERRY_LIBFUZZER==OFF`, `JERRY_LIBFUZZER==ON` and `JERRY_OSS_FUZZ==OFF`, and `JERRY_LIBFUZZER==ON` and `JERRY_OSS_FUZZ==ON`. This means that 3 possible settings are in the system. Therefore, I was suggesting to have ON, OSS, and OFF as valid values for `JERRY_LIBFUZZER` instead.

(The command line options for `tools/build.py` have a 1-to-1 correspondence to cmake options. So, the same goes for the `--libfuzzer` option of the script. It could have `['ON', 'OSS', 'OFF']` as valid choices.)

(Plus, if I'm not mistaken, `ON` and `OSS` are both "truthy" in cmake, that's why
```cmake
if(JERRY_LIBFUZZER)
  target_link_libraries("jerry-libfuzzer" jerry-port-default)
  if(NOT (JERRY_LIBFUZZER STREQUAL "OSS"))
    target_link_libraries("jerry-libfuzzer" -fsanitize=fuzzer)
  endif()
endif()
```
could simply work.)